src/main/java/socialite/model/person/containsKeywordsPredicate.java

```diff
@@ -0,0 +1,84 @@
+package socialite.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import socialite.commons.util.StringUtil;
+import socialite.model.tag.Tag;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
+public class containsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public containsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    //helper functions to test person's name, tags or handles respectively.
+    private boolean testName(Person person, String keyword) {
+        return StringUtil.containsWordIgnoreCase(person.getName().fullName, keyword);
+    }
+
+    private boolean testTags(Person person, String keyword) {
+        for (Tag tag : person.getTags()) {
+            if (StringUtil.containsWordIgnoreCase(tag.tagName, keyword)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean testPlatforms(Person person, String keyword) {
+
+        switch (keyword) {
+        case "facebook":
+            return !person.getFacebook().value.equals("");
+        case "instagram":
+            return !person.getInstagram().value.equals("");
+        case "telegram":
+            return !person.getTelegram().value.equals("");
+        case "tiktok":
+            return !person.getTiktok().value.equals("");
+        case "twitter":
+            return !person.getTwitter().value.equals("");
+        default:
+            return false;
```

Comment:
Wanted to say that we might consider using enum, but in hindsight I think this _**should**_ work okay, if the user type wrongly then bo bian they just type again. If other team members have better solution can suggest also!

Reply:
I have implemented enum in my PR already. Perhaps we can implement it here after we merge both PRs in. 

Although not sure how enums would work here as the switch variable is the keyword that the user typed in if i'm not wrong. And somehow the input would have to be mapped to the enum. Perhaps @david-eom could elaborate abit?